### PR TITLE
Fix child_process tests

### PR DESCRIPTION
**Pull request by ThatOneBro**
Fixed up very broken tests from node, added a few utils in `tests/bun.js/our-assert` for testing, including `createDoneDotAll` and `createCallCheckCtx` both very useful for working with callbacks and specifically event emitter callbacks.

### CODE REVIEW

File: src/bun.js/child_process.exports.js [hunk 1]
```diff
@@ -890,6 +894,7 @@ export class ChildProcess extends EventEmitter {
       err.spawnargs = ArrayPrototypeSlice.call(this.spawnargs, 1);
       this.emit("error", err);
     } else {
+      await this.#handleExited;
```
Comment (Jarred-Sumner) [inline src/bun.js/child_process.exports.js]: `exited` will return a resolved Promise when already exited

This is a good usecase for `Bun.peek`.

```suggestion
    const maybeExited = Bun.peek(this.#handleExited);
    if (maybeExited === this.#handleExited) {
       this.exitCode = await this.#handleExited;
    } else {
       this.exitCode = maybeExited;
    }
       
```

File: test/bun.js/our-assert.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
+import { expect } from "bun:test";
```
Comment (Jarred-Sumner) [inline test/bun.js/our-assert.js]: instead of `our-assert`, maybe `node-test-helpers`?

Comment (ThatOneBro) [reply test/bun.js/our-assert.js]: Yeah that makes sense

File: src/bun.js/child_process.exports.js (hunk 1 above)
Comment (ThatOneBro) [reply src/bun.js/child_process.exports.js]: Sick, I gotta get into the habit of thinking about Bun.peek use cases 🔥

**Comment by Jarred-Sumner** [comment]
Thank you

**Pull request merged**